tests/RegistrationForm.test.jsx

```diff
@@ -23,12 +30,38 @@ describe('RegistrationForm', () => {
         expect(passwordInput).toHaveAttribute('type', 'password')
         expect(passwordInput).toBeRequired()
 
-        const registerButton = screen.getByRole('button', { name: /register/i })
+        const registerButton = ( await screen.findAllByRole('button')).filter(button => button.textContent === 'Register')[0]
```

Comment:
If you install the https://chromewebstore.google.com/detail/testing-playground/hejbmebodbijjdhflfknehhcgaklhano?pli=1 browser extension and use it on the site, it can give you a really good query. This one looks a bit complicated (good job to make it work tho), but probably will be hard to maintain.